src/main/java/org/skriptlang/skript/bukkit/itemcomponents/equippable/elements/ExprEquipCompEntities.java

```diff
@@ -0,0 +1,120 @@
+import org.skriptlang.skript.bukkit.itemcomponents.equippable.EquippableWrapper;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@SuppressWarnings("rawtypes")
+@Name("Equippable Component - Allowed Entities")
+@Description("The entities allowed to wear the item. "
+	+ "NOTE: Equippable component elements are experimental. Thus, they are subject to change and may not work as intended.")
+@Example("set the allowed entities of {_item} to a zombie and a skeleton")
+@Example("""
+	set {_component} to the equippable component of {_item}
+	clear the allowed entities of {_component}
+	""")
+@RequiredPlugins("Minecraft 1.21.2+")
+@Since("INSERT VERSION")
+public class ExprEquipCompEntities extends PropertyExpression<EquippableWrapper, EntityData> implements EquippableExperiment {
+
+	static {
+		registerDefault(ExprEquipCompEntities.class, EntityData.class, "allowed entities", "equippablecomponents");
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		//noinspection unchecked
+		setExpr((Expression<EquippableWrapper>) exprs[0]);
+		return true;
+	}
+
+	@Override
+	protected EntityData @Nullable [] get(Event event, EquippableWrapper[] source) {
+		List<EntityData> types = new ArrayList<>();
+		for (EquippableWrapper wrapper : source) {
+			EquippableComponent component = wrapper.getComponent();
+			Collection<EntityType> allowed = component.getAllowedEntities();
+			if (allowed == null || allowed.isEmpty())
+				continue;
+			allowed.forEach(entityType -> {
+				Class<? extends Entity> entityClass = entityType.getEntityClass();
+				types.add(EntityData.fromClass(entityClass));
+			});
+		}
+		return types.toArray(EntityData[]::new);
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		return switch (mode) {
+			case SET, DELETE, REMOVE, ADD -> CollectionUtils.array(EntityData[].class);
+			default -> null;
+		};
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		List<EntityType> converted = new ArrayList<>();
+		if (delta != null) {
+			for (Object object : delta) {
+				if (object instanceof EntityData<?> entityData)
+					converted.add(EntityUtils.toBukkitEntityType(entityData));
+			}
+		}
+
+		getExpr().stream(event).forEach(wrapper -> wrapper.editComponent(component -> {
+			Collection<EntityType> allowed = component.getAllowedEntities();
+			List<EntityType> current = allowed != null ? new ArrayList<>(allowed) : new ArrayList<>();
+			switch (mode) {
+				case SET -> component.setAllowedEntities(converted);
+				case ADD -> {
+					current.addAll(converted);
+					component.setAllowedEntities(current);
+				}
+				case REMOVE -> {
+					current.removeAll(converted);
+					component.setAllowedEntities(current);
+				}
+				case DELETE -> component.setAllowedEntities(new ArrayList<>());
```

Comment:
```suggestion
				case DELETE -> component.setAllowedEntities(null);
```
I believe the setter is nullable. Though, there is still wasteful list creation from `current` assignment